dev/TeachingTip/TeachingTip.cpp

```diff
@@ -242,164 +233,217 @@ bool TeachingTip::UpdateTail()
+            return std::make_tuple(firstColumnWidth, secondColumnWidth, nextToLastColumnWidth, lastColumnWidth);
+        }
+        return std::make_tuple(0.0f, 0.0f, 0.0f, 0.0f);
+    }();
 
-    float firstRowHeight = rowDefinitions.Size() > 0 ? static_cast<float>(rowDefinitions.GetAt(0).ActualHeight()) : 0.0f;
-    float secondRowHeight = rowDefinitions.Size() > 1 ? static_cast<float>(rowDefinitions.GetAt(1).ActualHeight()) : 0.0f;
-    float nextToLastRowHeight = rowDefinitions.Size() > 1 ? static_cast<float>(rowDefinitions.GetAt(rowDefinitions.Size() - 2).ActualHeight()) : 0.0f;
-    float lastRowHeight = rowDefinitions.Size() > 0 ? static_cast<float>(rowDefinitions.GetAt(rowDefinitions.Size() - 1).ActualHeight()) : 0.0f;
+    auto const [firstRowHeight, secondRowHeight, nextToLastRowHeight, lastRowHeight] = [this, nullableTailOcclusionGrid]()
+    {
+        if (auto const rowDefinitions = nullableTailOcclusionGrid ? nullableTailOcclusionGrid.RowDefinitions() : nullptr)
+        {
+            auto const firstRowHeight = rowDefinitions.Size() > 0 ? static_cast<float>(rowDefinitions.GetAt(0).ActualHeight()) : 0.0f;
+            auto const secondRowHeight = rowDefinitions.Size() > 1 ? static_cast<float>(rowDefinitions.GetAt(1).ActualHeight()) : 0.0f;
+            auto const nextToLastRowHeight = rowDefinitions.Size() > 1 ? static_cast<float>(rowDefinitions.GetAt(rowDefinitions.Size() - 2).ActualHeight()) : 0.0f;
+            auto const lastRowHeight = rowDefinitions.Size() > 0 ? static_cast<float>(rowDefinitions.GetAt(rowDefinitions.Size() - 1).ActualHeight()) : 0.0f;
+
+            return std::make_tuple(firstRowHeight, secondRowHeight, nextToLastRowHeight, lastRowHeight);
+        }
+        return std::make_tuple(0.0f, 0.0f, 0.0f, 0.0f);
+    }();
 
     UpdateSizeBasedTemplateSettings();
 
     switch (m_currentEffectiveTailPlacementMode)
     {
     // An effective placement of auto means the tip should not display a tail.
     case winrt::TeachingTipPlacementMode::Auto:
-        if (SharedHelpers::IsRS5OrHigher())
+        if (winrt::IUIElement9 tailOcclusionGridUIElement9 = nullableTailOcclusionGrid)
         {
-            tailOcclusionGrid.CenterPoint({ width / 2, height / 2, 0.0f });
+            tailOcclusionGridUIElement9.CenterPoint({ width / 2, height / 2, 0.0f });
         }
         UpdateDynamicHeroContentPlacementToTop();
         winrt::VisualStateManager::GoToState(*this, L"Untargeted"sv, false);
         break;
 
     case winrt::TeachingTipPlacementMode::Top:
-        if (SharedHelpers::IsRS5OrHigher())
+        if (winrt::IUIElement9 tailOcclusionGridUIElement9 = nullableTailOcclusionGrid)
+        {
+            tailOcclusionGridUIElement9.CenterPoint({ width / 2, height - lastRowHeight, 0.0f });
+        }
+        if (winrt::IUIElement9 tailEdgeBorderUIElement9 = m_tailEdgeBorder.get())
```

Comment:
Seeing this pattern a lot. Maybe add a TrySetCenterPoint(winrt::IUIElement9, Vector2) helper function in this file?